funding/manager.go

```diff
@@ -958,7 +971,33 @@ func (f *Manager) stateStep(channel *channeldb.OpenChannel,
 	// fundingLocked was sent to peer, but the channel was not added to the
 	// router graph and the channel announcement was not sent.
 	case fundingLockedSent:
-		err := f.addToRouterGraph(channel, shortChanID)
+		// in case we skipped confirmation, now that the channel is fully operational
+		// we still want to wait for the first confirmation to update the real short
+		// channel id.
+		var err error
+		if shortChanID.IsFake() {
+			// Adding the fake channel to the router graph
+			graph := f.cfg.Wallet.Cfg.Database.ChannelGraph()
+			fakeID := shortChanID
+			if err = f.addToRouterGraph(channel, shortChanID); err != nil {
+				return fmt.Errorf("failed adding to "+
+					"router graph: %v", err)
+			}
+			// Wait for confirmation and update the real short channel id.
+			if shortChanID, err = f.updateShortChannelID(channel); err != nil {
+				return fmt.Errorf("error waiting for "+
+					"funding confirmation: %v", err)
+			}
+			// Delete the fake channel from the graph.
+			if err = graph.DeleteChannelEdges(false, fakeID.ToUint64()); err != nil {
+				return fmt.Errorf("failed to remove fake channel from "+
+					"the graph: %v", err)
+			}
+			if err := f.fakeIDSManager.releaseFakeID(*fakeID); err != nil {
+				return fmt.Errorf("failed to release fake id %v", err)
+			}
+		}
+		err = f.addToRouterGraph(channel, shortChanID)
```

Comment:
> I think this should return early (the above statement), as otherwise, we end up calling `addToRouterGraph` _twice_.

We want  to call 'addToRouterGraph' twice actually once for the fake channel and the second time after the short channel id is changed and channel is confirmed.